### PR TITLE
chore(C): add breadth-first search and depth-first search

### PR DESCRIPTION
**Pull request by AnishLohiya**
**PR Checklist:**
- [X] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [X] My addition does not have a spelling problem
- [X] My submission has a proper and user-friendly description of the algorithm
- [X] My submission has the time complexity of the algorithm
- [ ] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [X] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

Added Breadth-first search and Depth First Search Algorithm in C. 

**Other information:**

### CODE REVIEW

**Comment by UG-SEP** [comment]
Sorry @ming-tsai I can't check it right know my exam are going on
Please give to another reviewer
🙏

**Review by Ashborn-SM — CHANGES_REQUESTED** [review]

File: algorithms/C/graphs/DFS.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    static int visited[5] = {0};
+    int j;
+    if (visited[start] == 0)
+    {
+        printf("%d ", start);
+        visited[start] = 1;
+        for (j = 1; j < n; j++)
+        {
+            if (G[start][j] == 1 && visited[j] == 0)
+                DFSTraversal(G, j, n);
```
Comment (Ashborn-SM) [inline algorithms/C/graphs/DFS.c:28]: missing out brackets could possibly lead to bug that would be difficult to find.
```suggestion
            if (G[start][j] == 1 && visited[j] == 0){
                DFSTraversal(G, j, n);
            }
```

**Review by Ashborn-SM — CHANGES_REQUESTED** [review]
Don't use acronym and capital letters, follow the [naming convention](https://github.com/MakeContributions/DSA/blob/main/README.md#1-contribution-specifications) used in the repo.
Also update the `README.md`

**Review by Ashborn-SM — CHANGES_REQUESTED** [review]
Fix the typo in the name of the file

Change `breath-first-search.c` `breadth-first-search.c`

**Review by Ashborn-SM — COMMENTED** [review]
Write the output of the program at the end of the file

**Review by Ashborn-SM — APPROVED** [review]
Good Work!!

**Pull request merged**